client/me/notification-settings/settings-form/device-selector.jsx

```diff
@@ -1,40 +1,33 @@
-import { size, map } from 'lodash';
 import PropTypes from 'prop-types';
-import { PureComponent } from 'react';
-import { connect } from 'react-redux';
 import FormSelect from 'calypso/components/forms/form-select';
-import getUserDevices from 'calypso/state/selectors/get-user-devices';
+import { useUserDevicesQuery } from 'calypso/data/user-devices/use-user-devices-query';
 import StreamHeader from './stream-header';
 
-class NotificationSettingsFormDeviceSelector extends PureComponent {
-	static propTypes = {
-		devices: PropTypes.array.isRequired,
-		selectedDeviceId: PropTypes.number.isRequired,
-		onChange: PropTypes.func.isRequired,
-	};
+function NotificationSettingsFormDeviceSelector( { selectedDeviceId, onChange } ) {
+	const { data: devices = [] } = useUserDevicesQuery();
 
-	render() {
-		const { devices } = this.props;
-		if ( size( devices ) === 1 ) {
-			return <StreamHeader title={ devices[ 0 ].name } />;
-		}
+	if ( devices.length === 1 ) {
+		return <StreamHeader title={ devices[ 0 ].name } />;
+	}
 
-		return (
-			<div className="notification-settings-form-header">
-				<div className="notification-settings-form-header__title">
-					<FormSelect value={ this.props.selectedDeviceId } onChange={ this.props.onChange }>
-						{ map( devices, ( { id, name } ) => (
-							<option key={ id } value={ id }>
-								{ name }
-							</option>
-						) ) }
-					</FormSelect>
-				</div>
+	return (
+		<div className="notification-settings-form-header">
+			<div className="notification-settings-form-header__title">
+				<FormSelect value={ selectedDeviceId } onChange={ onChange }>
+					{ devices.map( ( { id, name } ) => (
+						<option key={ id } value={ id }>
```

Comment:
I'd suggest prefixing the `key` with something kere:

```suggestion
						<option key={ `device-notifications-${ id }` } value={ id }>
```